### PR TITLE
fix(scylla_bench_thread): add all available node to command

### PR DESCRIPTION
**Pull request by fruch**
Since in some cases when the stress command start, one of the
nodes might be down, and we need to make sure the stress command
can connect to any of the available nodes and start the stress.

Fixes: #4386

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/scylla_bench_thread.py
Comment (vponomaryov) [inline sdcm/scylla_bench_thread.py:180]: Shouldn't lines 180, 185 and 186 be changed too? These are node-specific.

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: sdcm/scylla_bench_thread.py
Comment (fruch) [reply sdcm/scylla_bench_thread.py:180]: in those lines, the node is the loader.  (using cql_ip_address is not correct here, but not breaking anything)

**Review by vponomaryov — DISMISSED** [review]
LGTM

File: sdcm/scylla_bench_thread.py
Comment (juliayakovlev) [inline sdcm/scylla_bench_thread.py:178]: @fruch did you try to run? AFAIR scylla-bench is not able to run on few nodes. But maybe it was changed?

**Comment by fruch** [comment]
Testing this is:
https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-large-partition-asymmetric-cluster-3h-test/3/

File: sdcm/scylla_bench_thread.py
Comment (fruch) [reply sdcm/scylla_bench_thread.py:178]: seem like it's working just fine

Comment (fruch) [reply sdcm/scylla_bench_thread.py:178]:
```
Command "/$HOME/go/bin/scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=25 -clustering-row-count=1     -partition-offset=476 -clustering-row-size=uniform:8192..10240  -concurrency=10  -connection-count=10  -consistency-level=quorum -rows-per-request=10 -timeout=30s -iterations 10 -error-at-row-limit 1000 -nodes 10.0.1.204,10.0.1.72,10.0.2.215,10.0.1.70,10.0.2.102" finished with status 0
```

**Comment by fruch** [comment]
> Testing this is: https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-large-partition-asymmetric-cluster-3h-test/3/

job running o.k. (failing cause of scylla-bench timing out, but that's a known issue)

**Comment by juliayakovlev** [comment]
> > Testing this is: https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-large-partition-asymmetric-cluster-3h-test/3/
> 
> job running o.k. (failing cause of scylla-bench timing out, but that's a known issue)

Please, see this PR. https://github.com/scylladb/scylla-cluster-tests/pull/1810
I'll check your monitor

**Comment by juliayakovlev** [comment]
> > > Testing this is: https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-large-partition-asymmetric-cluster-3h-test/3/
> > 
> > 
> > job running o.k. (failing cause of scylla-bench timing out, but that's a known issue)
> 
> Please, see this PR. #1810 I'll check your monitor

The load looks unbalanced on the nodes:

Screenshot from 2022-06-21 07-44-25

**Comment by fruch** [comment]
> > > > Testing this is: https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-large-partition-asymmetric-cluster-3h-test/3/
> > > 
> > > 
> > > job running o.k. (failing cause of scylla-bench timing out, but that's a known issue)
> > 
> > 
> > Please, see this PR. #1810 I'll check your monitor
> 
> The load looks unbalanced on the nodes:
> 
> Screenshot from 2022-06-21 07-44-25

Well this cases uses different shard number for each node, so it's logical that the load isn't even.

I still fail to understand how selecting multiple nodes as contact point changes any logic, if it does it's a but in gocql driver

**Comment by juliayakovlev** [comment]
> > > > > Testing this is: https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-large-partition-asymmetric-cluster-3h-test/3/
> > > > 
> > > > 
> > > > job running o.k. (failing cause of scylla-bench timing out, but that's a known issue)
> > > 
> > > 
> > > Please, see this PR. #1810 I'll check your monitor
> > 
> > 
> > The load looks unbalanced on the nodes:
> > Screenshot from 2022-06-21 07-44-25
> 
> Well this cases uses different shard number for each node, so it's logical that the load isn't even.
> 
> I still fail to understand how selecting multiple nodes as contact point changes any logic, if it does it's a but in gocql driver

AFAIR we run different configuration and got conclusion that this behaviour of (yes) gocql driver.
It was more then 2 year ago and maybe the behaviour has been changed. I suggest to run regular test (not asymm) and compare the results before merging

**Comment by fruch** [comment]
> AFAIR we run different configuration and got conclusion that this behaviour of (yes) gocql driver. It was more then 2 year ago and maybe the behaviour has been changed. I suggest to run regular test (not asymm) and compare the results before merging

this is from the same case running with only one node in the s-b commandline:

image

looks the same to me.... (I think s-b is usage of different partitions isn't balanced to begin with)

**Comment by fruch** [comment]
> AFAIR we run different configuration and got conclusion that this behaviour of (yes) gocql driver. It was more then 2 year ago and maybe the behaviour has been changed. I suggest to run regular test (not asymm) and compare the results before merging


taking it for a final test:
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-large-partition-8h-test/3/

**Comment by fruch** [comment]
@roydahan what do you think about doing this change across the whole stress threads ?

I would think we should by default pass all the nodes to all the stress command, since the drivers would connect to them all anyhow. if it would be causing issues, it's actually a driver bug (or scylla).

**Comment by fruch** [comment]
taking it for a ride again, with newer driver version
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-large-partition-8h-test/4/

**Comment by fruch** [comment]
> taking it for a ride again, with newer driver version https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-large-partition-8h-test/4/

results seems more balanced when using newer driver and workaround a host selection bug in the driver
image

https://github.com/gocql/gocql/issues/1621
https://github.com/scylladb/scylla-bench/pull/97

**Comment by fruch** [comment]
my conclusion the the imbalanced report before would be fixed by
https://github.com/scylladb/scylla-bench/pull/97

selecting multiple contact points isn't harming/effect the situation (that we are trying to drill down to with @vponomaryov on a different thread)

@roydahan @vponomaryov, waiting for you review on this one.

**Review by vponomaryov — APPROVED** [review]
My testing showed that there is no diff having either one or all node IP addresses in the stress command.
If you think that if may help somehow then let it be.

**Comment by fruch** [comment]
> My testing showed that there is no diff having either one or all node IP addresses in the stress command. If you think that if may help somehow then let it be because I also don't see any harm having full list of IPs.

it would help in addressing #4386 only, not any other issue we've seen with scylla-bench

**Pull request merged**
